petastorm/tf_utils.py

```diff
@@ -69,13 +71,20 @@ def _sanitize_field_tf_types(sample):
             next_sample_dict[k] = str(v.normalize())
         elif isinstance(v, np.ndarray) and np.issubdtype(v.dtype, np.datetime64):
             # Convert to nanoseconds from POSIX epoch
-            next_sample_dict[k] = (v - np.datetime64('1970-01-01T00:00:00.0Z')) \
+            next_sample_dict[k] = (v - np.datetime64('1970-01-01T00:00:00.0')) \
                 .astype('timedelta64[ns]').astype(np.int64)
         elif isinstance(v, np.ndarray) and v.dtype == np.uint16:
             next_sample_dict[k] = v.astype(np.int32)
         elif isinstance(v, np.ndarray) and v.dtype.type in (np.bytes_, np.unicode_):
             if v.size != 0:
                 next_sample_dict[k] = v.tolist()
+        elif isinstance(v, np.ndarray) and v.dtype.kind == 'O' and isinstance(v[0], datetime.date):
+            # Pyarrow 0.12.1 started returning python datetime.date when parquet column is a DateType() column.
+            # Convert values in such column into nsec from epoch int64.
+            def date_to_nsec_from_epoch(dt):
+                return timegm(dt.timetuple()) * 1000000000
+            to_nsec = np.vectorize(date_to_nsec_from_epoch)
```

Comment:
This is executed for every value. Since this is constant, it would make sense to define it outside of this function